drivers/soft_spi/soft_spi.c

```diff
@@ -137,29 +138,33 @@ void soft_spi_release(soft_spi_t bus)
 
 static inline uint8_t _transfer_one_byte(soft_spi_t bus, uint8_t out)
 {
-    int8_t bit = 0, i = 0;
+    int8_t i = 8;
     if (SOFT_SPI_MODE_1 == soft_spi_config[bus].soft_spi_mode ||
         SOFT_SPI_MODE_3 == soft_spi_config[bus].soft_spi_mode) {
         /* CPHA = 1*/
         gpio_toggle(soft_spi_config[bus].clk_pin);
     }
 
-    bit = (out & (1 << 7)) >> 7;
-    gpio_write(soft_spi_config[bus].mosi_pin, bit);
-    for (i = 6; i >= 0; i--) {
+    do{
+        int8_t bit = (out & (1 << 7)) >> 7;
+        gpio_write(soft_spi_config[bus].mosi_pin, bit);
+
         xtimer_nanosleep(soft_spi_config[bus].soft_spi_clk);
         gpio_toggle(soft_spi_config[bus].clk_pin);
+
+        out <<= 1; /*shift transfer register*/
+
+        bit = gpio_read(soft_spi_config[bus].miso_pin);
+        out = bit ? out | 0x01 : out & 0xfe; /*set or delete bit 0*/
```

Comment:
```suggestion
        out = bit ? (out | 0x01) : (out & 0xfe); /*set or delete bit 0*/
```

This is personal taste, so feel free to ignore. But IMO the extra braces here ease reading.